todo-mvc-kitchensink/src/todoProcesses.ts

```diff
@@ -0,0 +1,135 @@
+	const count = get('/todoCount');
+	const todo = { label: get('/currentTodo').trim(), id };
+
+	return todo.label ? [
+		add(`/todos/${id}`, todo),
+		replace('/currentTodo', ''),
+		replace('/todoCount', count + 1)
+	] : [];
+}
+
+function setCurrentTodoCommand({ payload: [ currentTodo ] }: CommandRequest): PatchOperation[] {
+	return [
+		replace('/currentTodo', currentTodo)
+	];
+}
+
+function removeTodoCommand({ get, payload: [ id ]  }: CommandRequest): PatchOperation[] {
+	const todo = get(`/todos/${id}`);
+	const operations = [];
+
+	if (todo) {
+		const completedCount = todo.completed ? get('/completedCount') - 1 : get('/completedCount');
+		const todoCount = get('/todoCount') - 1;
+
+		operations.push(replace('/completedCount', completedCount));
+		operations.push(replace('/todoCount', todoCount));
+		operations.push(replace('/completed', completedCount === todoCount && todoCount > 0));
+		operations.push(remove(`/todos/${id}`));
+	}
+
+	return operations;
+}
+
+function toggleTodoCommand({ get, payload: [ id ] }: CommandRequest): PatchOperation[] {
+	const completed = !get(`/todos/${id}/completed`);
+	const completedCount = (completed ? 1 : -1) + get('/completedCount');
+	const todoCount = get('/todoCount');
+
+	return [
+		replace('/completedCount', completedCount),
+		replace(`/todos/${id}/completed`, completed),
+		replace('/completed', completedCount === todoCount)
+	];
+
+}
+
+function toggleTodosCommand({ get }: CommandRequest): PatchOperation[] {
+	const completed = !get('/completed');
+	const completedCount = completed ? get('/todoCount') : 0;
+	return [
+		replace('/completedCount', completedCount),
+		replace('/completed', completed),
+		...Object.keys(get('/todos')).map(key => replace(`/todos/${key}/completed`, completed))
+	];
+}
+
+function editTodoCommand({ payload: [ todo ] }: CommandRequest): PatchOperation[] {
+	return [ replace('/editedTodo', todo) ];
+}
+
+function clearCompletedCommand({ get }: CommandRequest): PatchOperation[] {
+	const todos: Todos = get('/todos');
+	const keys = Object.keys(todos);
+	const todoCount = get<number>('/todoCount') - get<number>('/completedCount');
+
+	return [
+		replace('/todoCount', todoCount),
+		replace('/completedCount', 0),
+		replace('/completed', false),
+		...keys.filter(key => todos[key].completed).map(key => remove(`/todos/${key}`))
+	];
+}
+
+function saveTodoCommand({ get }: CommandRequest): PatchOperation[] {
+	const editedTodo = get<Todo>('/editedTodo');
+
+	return editedTodo ? [
+		replace(`/todos/${editedTodo.id}`, editedTodo),
+		replace('/editedTodo', undefined)
+	] : [];
+}
+
+function searchCommand({ payload: [ search ] }: CommandRequest): PatchOperation[] {
+	return [ replace('/currentSearch', search) ];
+}
+
+function initialStateCommand() {
+	return [
+		add('/completedCount', 0),
+		add('/completed', false),
+		add('/currentSearch', ''),
+		add('/currentTodo', ''),
+		add('/editedTodo', undefined),
+		add('/todoCount', 0),
+		add('/todos', {})
```

Comment:
Why are the todos an object? presumably keyed by their id... It's probably a hangover from the previous implementation, I think we can change the app data of we think that it could be better structured.

Reply:
When I updated the other version of `todo-mvc` (on accident, read the issue wrong), it was using an array instead of an object. I actually think this is a little nicer to work with, but that's partially because we're updating the counts incrementally. If you want me to update it to separately calculate the counts after each command that modifies/updates/removes todos then it would probably be better to use an array instead.